### PR TITLE
Websocket connection: create class for handling websockets #2503

### PR DESCRIPTION
**Pull request by ashklianko**

### CODE REVIEW

File: src/main/resources/assets/admin/common/js/connection/WebSocketConnection.ts [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+    protected handleWSError(ev: ErrorEvent): void {
+        if (this.debug) {
+            console.error('WebSocketConnection: Unable to connect to server web socket on ' + this.wsUrl, ev);
+        }
+
+        this.notifyConnectionError();
+    }
+
+    protected handleWSMessage(wsMessage: any): void {
```
Comment (alansemenov) [inline src/main/resources/assets/admin/common/js/connection/WebSocketConnection.ts:161]: Change `any`

File: src/main/resources/assets/admin/common/js/event/ServerEventsConnection.ts [hunk 2]
```diff
@@ -4,205 +4,66 @@ import {UriHelper} from '../util/UriHelper';
...
-    private handleWSError(ev: ErrorEvent) {
-        if (this.debug) {
-            console.error('ServerEventsConnection: Unable to connect to server web socket on ' + this.wsUrl, ev);
-        }
-
-        this.notifyConnectionError();
-    }
-
-    private handleWSMessage(remoteEvent: any) {
+    protected handleWSMessage(remoteEvent: any): void {
```
Comment (alansemenov) [inline src/main/resources/assets/admin/common/js/event/ServerEventsConnection.ts:66]: Change `any`

**Review by alansemenov — CHANGES_REQUESTED** [review]

File: src/main/resources/assets/admin/common/js/connection/WebSocketConnection.ts [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    private stateChangeTime: number;
+
+    private connectionState: CONNECTION_STATE = CONNECTION_STATE.NOT_ESTABLISHED;
+
+    public debug: boolean = false;
+
+    constructor(builder: WebSocketConnectionBuilder) {
+        assertNotNull(builder.url, i18n('connection.websocket.url.missing'));
+
+        this.ws = null;
```
[the remark this answers is not in the document]
Comment (ashklianko) [reply src/main/resources/assets/admin/common/js/connection/WebSocketConnection.ts:44]: sorry, just blindly copied some parts from ServerEventsConnection+

**Comment by ashklianko** [comment]
Updated!

**Pull request merged**